src/challenge-2/challenge-2.c

```diff
@@ -1,29 +1,89 @@
-#include <json-c/json.h>
-#include <stdio.h>
-
 #include "../../lib/collections.h"
 #include "../../lib/util.h"
 #include "json-c/json_object.h"
+#include <json-c/json.h>
+#include <stdio.h>
+
+char* LEADER_NODE = "n0";
 
 json_object* init_reply(json_object* init_msg);
 
 void client_request_handler(json_object* client_request);
 void conch_response_handler(json_object* conch_response);
+void conch_request_handler(json_object* conch_request, Queue* conch_queue);
+void conch_release_handler(json_object* conch_release, Conch* conch);
+void conch_response_dispatcher(Conch* conch, Queue* request_queue);
 
-char* LEADER_NODE = "n0";
-
-void client_request_handler(json_object* client_request)
+int main(void)
 {
-
-    /* Conch request schema
+    json_object* init_msg = msg_recv();
+    if (init_msg == NULL)
     {
-      "src": "n_",
-      "dest": "n0",
-      "body": { "type": "conch_request", "client_id": "c_" }
+        fprintf(stderr, "expected init message, got EOF");
+        exit(EXIT_FAILURE);
     }
-    */
+    msg_send(init_reply(init_msg));
+    json_object_put(init_msg);
+
+    // TODO: Event loops
+}
+
+json_object* init_reply(json_object* init_msg)
+{
+    return generic_reply(init_msg);
+}
+
+// Used exclusively by Leader
+void conch_request_handler(json_object* conch_request, Queue* request_queue)
+{
+    json_object* body = json_object_object_get(conch_request, "body");
+    queue_enqueue(request_queue,
+                  json_object_object_get(body, "original_client_request"));
```

Comment:
I think you need a `_get` here for the refcount of the `original_client_request` json_object -- otherwise, it might get freed while the queue still holds the pointer to it.